handler.cpp

```diff
@@ -64,17 +64,41 @@ RequestHandler::Status StaticHandler::Init(const std::string& uri_prefix, const
             serve_path = config.statements_[i]->tokens_[1];
         }
     }
-
+    this->uri_prefix = uri_prefix;
     return RequestHandler::PASS;
 }
 
 // Attempts to serve file that was requested by the request object
 // If not possible, we return fail signal for outside class to handle
 // calling not found handler
 RequestHandler::Status StaticHandler::HandleRequest(const Request& request, Response* response) {
-    std::string abs_path = serve_path + "/" + request.uri();
-    std::cout << "Attempting to serve file from: " << abs_path << std::endl;
+    
+    std::string abs_path = serve_path;
+    //get the file name/path from the request url
+    std::string file_path = request.uri().substr(uri_prefix.length());
+
+    //Code to add the / between the file path and serve_path properly
+    if (file_path.length() != 0 && file_path[0] == '/'){
+        if (abs_path.length() != 0 && abs_path[abs_path.length()-1] == '/'){
+            if (abs_path.length() == 1)
+                abs_path == "";
+            else
+                abs_path = abs_path.substr(1);
+        }
+    }else{
+        if (abs_path.length() != 0 && abs_path[abs_path.length()-1] != '/'){
+            if (file_path.length() != 0)
+                abs_path += "/";
+        }
+    }
+
+    abs_path += file_path;
 
+    std::cout << "Attempting to serve file from: " << serve_path << std::endl;
+
+    //WARNING WE SHOULD LOOK AT THIS. MIGHT CAUSE AN ERROR BECAUSE
```

Comment:
I believe I have a solution to this:
http://stackoverflow.com/questions/328944/how-do-i-check-if-a-file-is-a-regular-file-in-c

I'm down to merge this them implement it, just a small change in the code of file_exists. I remember when I was implementing it boost::filesystem was giving me problems but it seems to be working for everyone else. Looks simple enough.